packages/syft/src/syft/core/tensor/smpc/mpc_tensor.py

```diff
@@ -251,26 +346,58 @@ def __apply_public_op(self, y: "MPCTensor", op_str: str) -> List[ShareTensor]:
 
     def __apply_op(
         self,
-        y: Union[int, float, torch.Tensor, np.ndarray, "MPCTensor"],
+        y: Union[int, float, torch.Tensor, np.ndarray, MPCTensor],
         op_str: str,
-    ) -> "MPCTensor":
+    ) -> MPCTensor:
         """Apply an operation on "self" which is a MPCTensor "y".
 
          This function checks if "y" is private or public value.
 
         Args:
-            y (Union[int, float, torch.Tensor, np.ndarray, "MPCTensor"]: tensor to apply the operation.
+            y (Union[int, float, torch.Tensor, np.ndarray, MPCTensor]: tensor to apply the operation.
             op_str (str): the operation.
 
         Returns:
             MPCTensor. the operation "op_str" applied on "self" and "y"
         """
-        is_private = isinstance(y, MPCTensor)
 
-        if is_private:
-            result = self.__apply_private_op(y, op_str)
+        _self = self
+        if ispointer(y):
+            client = getattr(y, "client", None)
+            if self.parties is None:
+                raise Exception("")
+            if client is not None and client not in self.parties:
+                parties = self.parties + [client]
+            else:
+                parties = [party for party in self.parties]
+
+            # TODO: Extract info for y shape from somewhere
```

Comment:
Does this need to be handled in the backlog?

Reply:
added